xla/service/gpu/autotuner_util.h

```diff
@@ -174,13 +182,22 @@ class AutotuneConfig {
 
   bool ExhaustiveTilingSearch() const { return exhaustive_tiling_search_; }
 
+  int shard_index() const { return shard_index_; }
+  int shard_count() const { return shard_count_; }
+  std::shared_ptr<KeyValueStoreInterface> key_value_store() const {
+    return key_value_store_;
+  }
+
  private:
   std::variant<DeviceConfig, DevicelessConfig> config_;
   int32_t autotune_level_;
   bool should_crash_on_check_failure_;
   bool exhaustive_tiling_search_;
   bool require_complete_aot_autotune_results_;
   mutable std::unique_ptr<se::DeviceMemoryAllocator> allocator_;
+  int shard_index_;
+  int shard_count_;
+  std::shared_ptr<KeyValueStoreInterface> key_value_store_;
```

Comment:
Might be a nit, but I'm not fully sure associating KV storage with the Config is correct, since config is meant to be a lightweight value type essentially describing a GPU. Maybe makes more sense to pass it around separately?

Reply:
done